doc/algorithms.rst

```diff
@@ -233,6 +264,12 @@ where the desired overload is selected based on the :cpp:`double` argument to th
 
 .. [#f1] In C++, the function signature corresponds to the function *declaration*, for which the type :cpp:`P` and :cpp:`P const` are treated identically by the compiler.
          However, for the function implementation or *definition*, algorithm authors are encouraged to use :cpp:`P const` to help guarantee the immutability of data.
+
+         .. admonition:: Chris Green
+            :class: admonition-chg
+
+            Should this have a reference to an explanation of this at e.g. ``cppreference.com``?
+
 .. [#f2] Phlex permits the registration of member functions that do not use the :cpp:`const` qualifier.
-         However, using such functions is highly discouraged as it indicates a class instance is modifiable during member-function execution, which is at odds with Phlex's functional-programming paradigm.
+         However, using such functions is highly discouraged as it indicates a class instance is modifiable during member-function execution, which is at odds with Phlex's functional-programming paradigm, and DUNE's stated desire for re[rpducibility.
```

Comment:
Functional programming can support reproducibility, but using a mutable class instance does not necessarily imply irreproducibility.  I think art-using people who read this proposed addition may object as art allows mutable modules that still can reproduce the same data products given the same conditions.